cpp/core/jni/JniWrapper.cc

```diff
@@ -229,7 +229,10 @@ jint JNI_OnLoad(JavaVM* vm, void* reserved) {
   metricsBuilderClass = createGlobalClassReferenceOrError(env, "Lorg/apache/gluten/metrics/Metrics;");
 
   metricsBuilderConstructor = getMethodIdOrError(
-      env, metricsBuilderClass, "<init>", "([J[J[J[J[J[J[J[J[J[JJ[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J)V");
+      env,
+      metricsBuilderClass,
+      "<init>",
+      "([J[J[J[J[J[J[J[J[J[JJ[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J)V");
```

Comment:
We eventually need a better way to pass metrics.

Reply:
Agreed. 

I have similar concern during working on a previous [patch](https://github.com/apache/incubator-gluten/pull/7842#discussion_r1865418672).

The code requires for a reworking.

Reply:
thanks @FelixYBW , agree with you.

Reply:
> We eventually need a better way to pass metrics.

@FelixYBW do you think "Foreign Function and Memory (FFM) API" from JDK 17 is a better way?

Reply:
@WangGuangxin I investigated the FFM API[1][2] and it seems that there is no way to create and call Java objects from the native side. Can you provide some information on this?

[1] https://docs.oracle.com/en/java/javase/22/core/foreign-function-and-memory-api.html
[2] https://openjdk.org/jeps/454

Reply:
FFM API/JNI's native side returns two-dimensional arrays, how about parsing mertrics on the java side?